### ISSUE TITLE
Split into two packages

### ISSUE THREAD

**Issue by bep** [natefinch/plugin#3]
I was looking for a plugin framework that does RPC over stdin/out. Looks promising, but ... 

It may feel tempting to have all in one package, but this feels really cluncky:

* `StartConsumer()`
* `StartProducer()`
* `NewConsumer()`
* `NewProducer()`
* ...

Compare it to:

* `consumer.Start()`
* `producer.Start()`
* ...

I'm guessing here, but from a usage perspective of this library, isn't it a Producer or a Consumer, never both at the same time?

**Comment by natefinch** [comment]
Yes, you can't have a plugin be both a provider and a consumer (it's defined by whether the plugin is the RPC server or client).

I had actually thought about splitting on the host/plugin sides.... the problem becomes that consumers, producers, and the main application all share code.  Plus, looking at the net/rpc library, it has both the client and the server in the same package.

Plus, in order to have them in separate packages, I'd have to put them in subfolders of the repo, which means the root folder of the repo would be basically useless.

While I do think that the naming is slightly better with separate packages, I don't think it's really that big of a benefit to justify the complexity added to the code.

**Comment by bep** [comment]
Sure, I haven't really thought it through.

**Comment by natefinch** [comment]
Oh, just noticed this is on the wrong repo... I've renamed the repo to https://github.com/natefinch/pie, and just haven't deleted this one yet.  I'm going to delete the repo now, but you're welcome to re-propose this over there if you want other people to see it.
